cmd/soroban-cli/src/commands/keys/add.rs

```diff
@@ -32,6 +36,16 @@ impl Cmd {
     pub fn run(&self, global_args: &global::Args) -> Result<(), Error> {
         let print = Print::new(global_args.quiet);
         let secret = self.secrets.read_secret()?;
+
+        if let Secret::SeedPhrase { seed_phrase } = &secret {
+            if seed_phrase.split_whitespace().count() < 24 {
```

Comment:
q: should be warn if seed phrase is neither 12/24 words?

Reply:
I think @leighmcculloch already handled that on https://github.com/stellar/stellar-cli/pull/1812.